### PR TITLE
[RESTEASY-3138] Remove dependency on commons-io.

### PR DESCRIPTION
**Pull request by jamezp**
https://issues.redhat.com/browse/RESTEASY-3138
https://issues.redhat.com/browse/RESTEASY-3139

### CODE REVIEW

**Comment by Thihup** [comment]
Would it be possible to use `InputStream.nullInputStream()` / `OutputStream.nullOutputStream()` in any of these cases?

**Comment by Thihup** [comment]
Would it be possible to use `InputStream.nullInputStream()` / `OutputStream.nullOutputStream()` in any of these cases?

**Comment by jamezp** [comment]
> Would it be possible to use `InputStream.nullInputStream()` / `OutputStream.nullOutputStream()` in any of these cases?

The only `NullInputStream` used was for a test and required an empty array to be initialized. I was hoping to just use `InputStream.nullInputStream()`, but the test would fail.

File: resteasy-client/src/main/java/org/jboss/resteasy/client/jaxrs/engines/EntityOutputStream.java [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+
+    /**
+     * The length of the length of the content written.
+     *
+     * @return the length of the content written
+     *
+     * @throws IOException if there is an error determining the length of the content
+     */
+    long getContentLength() throws IOException {
+        final Path file = this.file;
```
Comment (jbliznak) [inline resteasy-client/src/main/java/org/jboss/resteasy/client/jaxrs/engines/EntityOutputStream.java:198]: redundant line?

Comment (jamezp) [reply resteasy-client/src/main/java/org/jboss/resteasy/client/jaxrs/engines/EntityOutputStream.java:198]: Maybe I should lock, but the idea was to get the current value as its volatile and then use that below.

File: providers/multipart/pom.xml [hunk 2]
```diff
@@ -13,6 +13,10 @@
     <description/>
 
     <dependencies>
+        <dependency>
```
Comment (jbliznak) [inline providers/multipart/pom.xml:16]: is the order on first position deliberate?

File: resteasy-client/src/main/java/org/jboss/resteasy/client/jaxrs/engines/ApacheHttpAsyncClient4Engine.java [hunk 3]
```diff
@@ -878,4 +877,10 @@ private static IOException ioException(Exception ex) {
       return (ex instanceof IOException) ? (IOException) ex : new IOException(ex);
    }
 
+   private static void safeClose(final Closeable closeable) {
+      if (closeable != null) try {
```
Comment (jbliznak) [inline resteasy-client/src/main/java/org/jboss/resteasy/client/jaxrs/engines/ApacheHttpAsyncClient4Engine.java:881]: I know it is just matter of style but using `if` without brackets could be a bit cryptic, especially with try-catch

File: resteasy-client/src/main/java/org/jboss/resteasy/client/jaxrs/engines/EntityOutputStream.java (hunk 1 above)
Comment (jbliznak) [reply resteasy-client/src/main/java/org/jboss/resteasy/client/jaxrs/engines/EntityOutputStream.java:198]: I see, then lock seems safer if that matters, you already use plenty of that in this class.

**Pull request merged**